src/Language/PureScript/Sugar.hs

```diff
@@ -71,5 +71,5 @@ desugar externs =
     >=> rebracket externs
     >=> checkFixityExports
     >=> deriveInstances externs
-    >=> desugarTypeClasses externs
+    >=> (\(syns, kinds, mod) -> desugarTypeClasses externs syns kinds mod)
```

Comment:
Might I suggest pulling the synonym-finding out of `deriveInstances` and having it consumed by both stages, instead of this coupling of the two stages together? Then this section could be

```hs
    >=> checkFixityExports
    >=> (findTypeSynonyms externs >>= \syns ->
          deriveInstances externs syns
            >=> desugarTypeClasses externs syns)
    >=> createBindingGroupsModule
```

which keeps the pipeline flexible if stages get added or moved around.